### DISCUSSION TITLE
Duplicate beatmap when adding beatmap to collections

### DISCUSSION THREAD

**Discussion by sz-da** [ppy/osu#26590] [Q&A] [answered]
When adding a beatmap that has a duplicate to a collection, both of them will be added to the collection.

https://github.com/ppy/osu/assets/76718358/b935f4da-dc47-47d7-9604-a927b9edb8fb

Discussion instead of issue, as it is debatable whether the game should behave like this or not.

**Comment by peppy** [comment, marked as answer]
If they are the same beatmap, they are the same beatmap.

I'm not actually sure how you managed to import a duplicate like that (the game should be checking during import and disallow it) but this is not an issue with the collection logic.

**Reply by sz-da** [reply]
These 2 maps were imported from stable. Essentially it's a copy of a map with a different folder name (lots of those large beatmap packs/torrents have these kinds of duplicate maps). ~I know you mentioned before that duplicates are shown intentionally https://github.com/ppy/osu/issues/3059, but I honestly think it's better to hide or at least not import these duplicates from stable, as a lot of players unknowingly downloaded these duplicates from those map packs.~ Edit: seems like importing from stable has the same issue linked below, where the check during import fails for certain maps.

Doesn't apply to this map, but it is possible to import a duplicate map when importing osz files. I think you already noticed here https://github.com/ppy/osu/issues/26588

Will mark as answered though, as this discussion is referring to collections specifically.
